src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedTests.cs

```diff
@@ -0,0 +1,722 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+#nullable enable
+
+public class Test
+{{
+    public static void TestMethodConstrained([ConstantExpected(Min = {min}, Max = {outOfBoundMax})] {type} constant)
+    {{
+        TestMethodWithConstrainedConstant({{|#0:constant|}});
+    }}
+    public static void TestMethodWithConstrainedConstant([ConstantExpected(Min = {min}, Max = {max})] {type} val) {{ }}
+}}
+";
+            await TestCSAsync(csInput,
+                VerifyCS.Diagnostic(ConstantExpectedAnalyzer.CA1861.ConstantOutOfBoundsRule)
+                        .WithLocation(0)
+                        .WithArguments(minValue.ToString(), maxValue.ToString()));
+        }
+
+        [Fact]
+        public static async Task TestConstantCompositionNotSameTypeAsync()
+        {
+            string csInput = @"
+using System;
+using System.Diagnostics.CodeAnalysis;
+#nullable enable
+
+public class Test
+{
+    public static void TestMethod([ConstantExpected] long constant)
+    {
+        TestMethodWithConstant({|#0:(int)constant|});
+        TestMethodWithStringConstant({|#1:(string)(object)constant|});
+    }
+    public static void TestMethod([ConstantExpected] short constant)
+    {
+        TestMethodWithConstant({|#2:constant|});
+    }
+    public static void TestMethodWithConstant([ConstantExpected] int val) { }
+    public static void TestMethodWithStringConstant([ConstantExpected] string val) { }
+}
+";
+            await TestCSAsync(csInput,
+                VerifyCS.Diagnostic(ConstantExpectedAnalyzer.CA1861.ConstantInvalidConstantRule)
+                        .WithLocation(0)
+                        .WithArguments("int"),
+                VerifyCS.Diagnostic(ConstantExpectedAnalyzer.CA1861.ConstantInvalidConstantRule)
+                    .WithLocation(1)
+                    .WithArguments("string"),
+                VerifyCS.Diagnostic(ConstantExpectedAnalyzer.CA1861.ConstantInvalidConstantRule)
+                        .WithLocation(2)
+                        .WithArguments("int"));
+        }
+
+        private static async Task TestCSAsync(string source, params DiagnosticResult[] diagnosticResults)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = source,
+                LanguageVersion = CSharp.LanguageVersion.Preview,
+            };
+            // TODO: remove when the type is available
+            test.TestState.Sources.Add(s_attributeSource);
+            test.ExpectedDiagnostics.AddRange(diagnosticResults);
+            await test.RunAsync();
+        }
+
+        private static readonly string s_attributeSource = @"#nullable enable
+namespace System.Diagnostics.CodeAnalysis
+{
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
+    public sealed class ConstantExpectedAttribute : Attribute
+    {
+        public object? Min { get; set; }
+        public object? Max { get; set; }
+    }
+}";
```

Comment:
```suggestion
        private static readonly string s_attributeSource = @"
#nullable enable
namespace System.Diagnostics.CodeAnalysis
{
    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
    public sealed class ConstantExpectedAttribute : Attribute
    {
        public object? Min { get; set; }
        public object? Max { get; set; }
    }
}";
```

In general I suggested to remove all `#nullable enable` directives where I think they are not needed.
I am not sure if this is now enabled by default. In case it is enabled by default, I think we can remove it here as well.